File: Contract.Architecture.Frontends/Contract.Architecture.Web.Core/src/app/model/bankwesen/banken/dtos/i-bank-detail.ts
import { Kunde, IKunde } from 'src/app/model/kundenstamm/kunden/dtos/i-kunde';
import { ApiBankDetail } from './api/api-bank-detail';

export interface IBankDetail {
    id: string;
    name: string;
    boolean: boolean;
    dateTime: Date;
    double: number;
    guid: string;
    integer: number;
    kunden: IKunde[];
}

export abstract class BankDetail {
    public static fromApiBankDetail(apiBankDetail: ApiBankDetail): IBankDetail {
        return {
            id: apiBankDetail.id,
            name: apiBankDetail.name,
            boolean: apiBankDetail.boolean,
            dateTime: apiBankDetail.dateTime,
            double: apiBankDetail.double,
            guid: apiBankDetail.guid,
            integer: apiBankDetail.integer,
            kunden: apiBankDetail.kunden.map(apiKunde => Kunde.fromApiKunde(apiKunde)),
        };
    }
}

File: Contract.Architecture.Backends/Contract.Architecture.Backend.Core/Contract/Logic/Modules/Bankwesen/Banken/DTOs/IBankDetail.cs
using Contract.Architecture.Backend.Core.Contract.Logic.Modules.Kundenstamm.Kunden;
using System;
using System.Collections.Generic;

namespace Contract.Architecture.Backend.Core.Contract.Logic.Modules.Bankwesen.Banken
{
    public interface IBankDetail
    {
        Guid Id { get; set; }

        string Name { get; set; }

        bool Boolean { get; set; }

        DateTime DateTime { get; set; }

        double Double { get; set; }

        Guid Guid { get; set; }

        int Integer { get; set; }

        IEnumerable<IKunde> Kunden { get; set; }
    }
}